server/utils/FirebaseAlgoliaServer/functions/tests/cloudFunctionsIntegrationTests.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * BACKGROUND
+ * Integration tests for: 
+ * addEntityToIndex, deleteEntityFromIndex, updateRecordInIndex.
+ *
+ * These functions essentially link two different databases in realtime,
+ * And they cannot be directly tested because of how cloud functions work
+ * And because algolia does not provide a JS mocking library (AFAIK.)
+ * So testing has to be done in realtime online,
+ * Which makes testing somewhat difficult due to asynchronous operations
+ * Happening in weird orders.
+ * SO this code doesn't use unittests (couldn't get them to work), 
+ * and instead does live operations on the DBs, checks the DB state, 
+ * throws exceptions if something is amiss, 
+ * and if everything is ok will print to the console that things are working.
```

Comment:
Another issue:  If your test creates data, and then fails to remove it, is that data still there in the next test method? 
 How about in the next test run?

What would really be best is if you had test scaffolding to create a new DB each time you run this and dismantle it when it's done.  But in lieu of that you should at least verify before running each test that the state of the DB is as expected.  Or increment your read/write values to ensure that old data has no impact on a currently-running test.